src/main.rs

```diff
@@ -139,79 +147,162 @@ fn name_filter(font: &FontRef, needle: &Regex) -> bool {
         .any(|s| needle.is_match(&s.chars().collect::<String>()))
 }
 
-type StringFilter = dyn Fn(&FontRef, &str) -> bool;
+// Fast file extension check using byte-level comparison
+#[inline]
+fn is_font_file(name: &str) -> bool {
+    if name.ends_with(".otf") || name.ends_with(".ttf") {
```

Comment:
`if (condition) { return true } else { return false }` is usually written as `condition`. :-)